### PR TITLE
feat(p/json): optimize json

### PR DESCRIPTION
**Pull request by notJoon**
# Description

Optimized the JSON package and simplified JSON node creation using the builder pattern.

- in `buffer.gno` and `escape.gno` files are modified the use of map for lookup tables to use slice array instead.
- refactor the `Unquote` function in `escape.gno` file
- modified the existing functions that parsed numbers to use `strconv` package, and deleted related files and functions
    - especially, the `eisel_lemire` and `ryu` packages were deleted since they were files that had been added to handle `ParseUint` and `ParseFloat` in `strconv` package.

## JSON Generate Example

**Plain JSON**

```go
node := Builder().
	WithString("name", "Alice").
	WithNumber("age", 30).
	WithBool("is_student", false).
	Node()

value, err := Marshal(node)
if err != nil {
	t.Errorf("unexpected error: %s", err)
}

Output:
{"name":"Alice","age":30,"is_student":false}
```

**Nested Structure**

```go
node := Builder().
	WriteString("name", "Alice").
	WriteObject("address", func(b *NodeBuilder) {
		b.WriteString("city", "New York").
		WriteNumber("zipcode", 10001)
	}).
	Node()
// ...

Output:
{"name":"Alice","address":{"city":"New York","zipcode":10001}}
```

## Benchmark Result for Unquote

**Before**

```plain
BenchmarkUnquote-8              	12433488	        98.06 ns/op	     144 B/op	       2 allocs/op
BenchmarkUnquoteWorstCase-8     	24727736	        50.46 ns/op	      48 B/op	       1 allocs/op
BenchmarkUnquoteBestCase-8      	22542354	        52.69 ns/op	      48 B/op	       1 allocs/op
BenchmarkUnquoteEmptyString-8   	394868628	         3.067 ns/op	       0 B/op	       0 allocs/op
```

**After**

```plain
BenchmarkUnquote-8              	12464704	        96.61 ns/op	     144 B/op	       2 allocs/op
BenchmarkUnquoteWorstCase-8     	25084070	        48.02 ns/op	      48 B/op	       1 allocs/op
BenchmarkUnquoteBestCase-8      	23383227	        52.66 ns/op	      48 B/op	       1 allocs/op
BenchmarkUnquoteEmptyString-8   	400496838	         2.968 ns/op	       0 B/op	       0 allocs/op
```

### CODE REVIEW

**Review by jefft0 — APPROVED** [review]
This is follow-on PR by the same author of the original package in https://github.com/gnolang/gno/pull/1415 . On my macOS, all tests pass with `make test.components` . Removed the `review/triage-pending` label. Ready for review by core devs.

**Comment by notJoon** [comment]
> Hi @notJoon. Maybe it's obvious to people familiar with the code, but in the description can you put a short explanation why you removed the eisel_lemire and ryu folders?

description updated. thanks!

**Review by thehowl — APPROVED** [review]

**Comment by thehowl** [comment]
BTW the benchmarks are in go and are likely to mean very little in Gno ;)

At most, use `-print-runtime-metrics` for a comparison, even though it's still going to not be terribly useful to understand real-world CPU usage (yet).

File: examples/gno.land/p/demo/json/buffer.gno [hunk 1]
```diff
@@ -33,7 +33,7 @@ func (b *buffer) first() (byte, error) {
 	for ; b.index < b.length; b.index++ {
 		c := b.data[b.index]
 
-		if !(c == whiteSpace || c == carriageReturn || c == newLine || c == tab) {
```
[the remark this answers is not in the document]
Comment (notJoon) [reply examples/gno.land/p/demo/json/buffer.gno]: reverted this part as it seemed to unnecessarily increase dependencies. thanks! https://github.com/gnolang/gno/pull/2939/commits/f09e3399f7195d48a1e57e4ae950fc687b629a8f

**Pull request merged**